src/routes/modules/modules.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { ModuleStudents } from 'components/ModuleStudents/ModuleStudents';
+import { ModuleTeachers } from 'components/ModuleTeachers/ModuleTeachers';
+
+import { RootState } from 'store/store';
+import { setLoading, removeLoading } from 'store/loading/loadingSlice';
+import { updateToast, TOAST_GENERAL_ERRORS } from 'store/toast/toastSlice';
+
+import { UpdateModule, Module, StudentModule } from 'types/Module/Module';
+
+import { getClassModules, updateClassModules } from 'utils/db/db.utils';
+
+import style from './modules.module.css';
+
+export default function Modules() {
+  const dispatch = useDispatch();
+  const params = useParams();
+  const user = useSelector((state: RootState) => state.user.currentUser);
+
+  // Data from db. Modifiable by user
+  const [moduleData, setModuleData] = useState<(Module | StudentModule)[]>([]);
+  // To check if there are any changes on data
+  const [initialState, setInitialState] = useState<(Module | StudentModule)[]>(
+    []
+  );
+  const [editing, setEditing] = useState(false);
+  // Just to reload information from db
+  const [reload, setReload] = useState(0);
+
+  // Get information from db
+  useEffect(() => {
+    const getData = async () => {
+      dispatch(setLoading());
+      const response = await getClassModules(
+        params.id as string,
+        user?.authToken as string
+      );
+      if (response.status === 'success') {
+        setModuleData(response.data as (Module | StudentModule)[]);
+        setInitialState(response.data as (Module | StudentModule)[]);
+      } else {
+        dispatch(updateToast(TOAST_GENERAL_ERRORS.SYSTEM));
+      }
+      dispatch(removeLoading());
+    };
+    getData();
+  }, [reload, dispatch, params.id, user?.authToken]);
+
+  const hasChanges = moduleData.reduce((result, current, index) => {
+    if (current.is_active !== initialState[index].is_active) result = true;
+    return result;
+  }, false);
+
+  function onCancel() {
+    setEditing(false);
+    setModuleData(initialState);
+  }
+
+  async function onClickHandler() {
```

Comment:
I believe we had discussed in the past about using arrow functions for handler functions, but I just want to make sure

Reply:
![image](https://github.com/Code4UHub/client/assets/60989884/314f52fd-ab95-45c2-8a7a-852bba1eaf91)